src/Interpreters/ExpressionActionsSettings.h

```diff
@@ -9,16 +9,24 @@ namespace DB
 
 struct Settings;
 
+enum class CompileExpressions: uint8_t
```

Comment:
Interesting idea, but probably going to be annoying in practice -- have to remember these constants and can't use bool expressions. I just write `true /* compile expressions */`.

Reply:
> can't use bool expressions.

Agree, and BTW clang-tidy already has [`bugprone-argument-comment`](https://clang.llvm.org/extra/clang-tidy/checks/bugprone-argument-comment.html), that will report an error.

Reply:
@azat Let's enable it globally and fix it in all the codebase :)

Reply:
It is already enabled :)

https://github.com/ClickHouse/ClickHouse/blob/04e3ddeffa175c812cb4a311349e92f51b315a8c/.clang-tidy#L63

Reply:
It only warns on name mismatch but does not force to write a comment.

Reply:
I like emun class more than bool. Because it is impossible to miss argument position.